itest/performance/config.go

```diff
@@ -0,0 +1,136 @@
+package perf_test
+
+import (
+	"github.com/jessevdk/go-flags"
+)
+
+const (
+	// defaultConfigPath is the default path of the configuration file.
+	defaultConfigPath = "perf-test.conf"
+)
+
+// User defines the config options for a user in the network.
+type User struct {
+	Tapd *TapConfig `group:"tapd"  namespace:"tapd"`
+}
+
+// TapConfig are the main parameters needed for identifying and creating a grpc
+// client to a tapd subsystem.
+type TapConfig struct {
+	Name string `long:"name" description:"the name of the tapd instance"`
+	Host string `long:"host" description:"the host to connect to"`
+	Port int    `long:"port" description:"the port to connect to"`
+
+	TLSPath string `long:"tlspath" description:"Path to tapd's TLS certificate, leave empty if TLS is disabled"`
+	MacPath string `long:"macpath" description:"Path to tapd's macaroon file"`
+}
+
+// BitcoinConfig defines exported config options for the connection to the
+// btcd/bitcoind backend.
+type BitcoinConfig struct {
+	Host     string `long:"host" description:"bitcoind/btcd instance address"`
+	Port     int    `long:"port" description:"bitcoind/btcd instance port"`
+	User     string `long:"user" description:"bitcoind/btcd user name"`
+	Password string `long:"password" description:"bitcoind/btcd password"`
+	TLSPath  string `long:"tlspath" description:"Path to btcd's TLS certificate, if TLS is enabled"`
+}
+
+// Config holds the main configuration for the performance testing binary.
+type Config struct {
+	// TestCases is a comma separated list of test cases that will be
+	// executed.
+	TestCases string `long:"test-case" description:"the test case that will be executed"`
+
+	// Alice is the configuration for the main user in the network.
+	Alice *User `group:"alice" namespace:"alice" description:"alice related configuration"`
+
+	// Bob is the configuration for the secondary user in the network.
+	Bob *User `group:"bob" namespace:"bob" description:"bob related configuration"`
+
+	// Bitcoin is the configuration for the bitcoin backend.
+	Bitcoin *BitcoinConfig `group:"bitcoin" namespace:"bitcoin" long:"bitcoin" description:"bitcoin client configuration"`
+
+	// BatchSize is the number of assets to mint in a single batch. This is only
+	// relevant for some test cases.
+	BatchSize int `long:"batch-size" description:"the number of assets to mint in a single batch"`
+
+	// TestSuiteTimeout is the timeout for the entire test suite in minutes.
+	TestSuiteTimeout int `long:"test-suite-timeout" description:"the timeout for the entire test suite in minutes"`
```

Comment:
You can make both this and the one below a `time.Duration`. Then users can specify `5m`, `1s`, etc.